server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -249,10 +249,17 @@ public void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueExcept
 
     public long computeNextDeliveryTimestamp(long delay, TimeUnit unit) {
         if (delay > 0) {
-            return ZonedDateTime.now()
-                .plus(delay, Temporals.chronoUnit(unit))
-                .toInstant()
-                .toEpochMilli();
+            try {
+                return ZonedDateTime.now()
+                    .plus(delay, Temporals.chronoUnit(unit))
+                    .toInstant()
+                    .toEpochMilli();
+            } catch (ArithmeticException e) {
+                LOGGER.warn("The {} was caused by conversation {}({}) followed by addition to current timestamp. Falling back to Long.MAX_VALUE.",
+                        e.getMessage(), delay, unit.name());
```

Comment:
Yes, seem logical. Because we use `Long.MAX` and not every person (for example me) would recognize it's decimal value (`9223372036854775807`) it might be unconvenient/confusing to read that message, so I suggest to use `Fallback to {} (Long.MAX) ms`.

This message also good to explicitly indicate that we use maximum value. For reader there is not deduction process that he might make.

Reply:
> so I suggest to use `Fallback to {} (Long.MAX) ms`

I'm also part of these primitive human beings I guess, then :+1: seems great!